src/simulation/m_bubbles.fpp

```diff
@@ -38,37 +38,45 @@ contains
         !!  @param f_bub_adv_src Source for bubble volume fraction
         !!  @param f_divu Divergence of velocity
         !!  @param fCson Speed of sound from fP (EL)
-    function f_rddot(fRho, fP, fR, fV, fR0, fpb, fpbdot, alf, fntait, fBtait, f_bub_adv_src, f_divu, fCson)
+        !!  @param gam_l Physical Bubble parameter (pass in from global parameters)
+        !!  @param Ca_l Cavitation number (pass in from global parameters)
+        !!  @param Web_l Weber number (pass in from global parameters)
+        !!  @param Re_inv_l Inverse Reynolds number (pass in from global parameters)
+        !!  @param bubbles_euler_l Bubbles euler on/off (pass in from global parameters)
+        !!  @param polytropic_l Polytropic  switch (pass in from global parameters)
+    pure elemental function f_rddot(fRho, fP, fR, fV, fR0, fpb, fpbdot, alf, fntait, fBtait, f_bub_adv_src, f_divu, fCson, gam_l, Ca_l, Web_l, Re_inv_l, bubbles_euler_l, polytropic_l)
         !$acc routine seq
         real(wp), intent(in) :: fRho, fP, fR, fV, fR0, fpb, fpbdot, alf
         real(wp), intent(in) :: fntait, fBtait, f_bub_adv_src, f_divu
         real(wp), intent(in) :: fCson
+        real(wp), intent(in) :: gam_l, Ca_l, Web_l, Re_inv_l
+        logical, intent(in) :: bubbles_euler_l, polytropic_l
```

Comment:
Was adding these as dummy arguments necessary for the compiler to accept this as a pure elemental procedure? Since these can be made `intent(in)` they must not have been modified prior to the refactoring, and so the routine should have had no side effects.

Reply:
see my comment on slack: 
`i’m curious what you think is a good strategy here. i think was the one who recommended passing in globals as parameters to a subroutine. i’m not sure if this is required to for a subroutine to be pure or not. it does seem that if they aren’t passed with intent(in) then they could be changed within the routine and thus cause side effects?
for what it’s worth - this is partially an effort to use fewer module globals and make subroutines more self-contained, but maybe that’s orthogonal to making things pure or elemental? thoughts?`